lib/likelihood.c

```diff
@@ -121,12 +121,13 @@ msp_log_likelihood_arg(tsk_treeseq_t *ts, double r, double Ne, double *r_lik)
     double material = lineages * tsk_treeseq_get_sequence_length(ts);
     double material_in_children, material_in_parent, rate, gap;
     double lik = 0;
+    double left_end, right_end;
```

Comment:
Did you intend to include these changes in the likelihood calculation? I thought the idea was to leave this to a follow-up?

Reply:
Yes. This works whenever there isn't an offending diamond in the ARG, and is as far as I got before realising the problem. I thought it would be quickest to include it here to have as a starting point for the fix once the metadata is in. Happy to do something different too if you think that's not good practice.

Reply:
I think it probably is a good idea to back out of these changes on this branch, and try to keep them separate.

The easiest thing to do is to start a new branch based on your current one, which can be the starting point for likelihood updates. You can then reset the state of the likelihood files to what they are in upstream/main on this branch and it should all go smoothly.

Reply:
Sounds good to me. I'll push one more update to that effect and ping you.